### PR TITLE
fix(InMemoryProvider): updateGuild and updateAxon return values

### PR DESCRIPTION
**Pull request by bsian03**
- Error and kill process if no token supplied
 - Return correct value in dbprovider methods

### CODE REVIEW

**Comment by Khaazz** [comment]
It seems you indeed have noticed an inconsistency here.
But the return type for updateGuild and updateAxon is actually correct.
The way the framework is currently designed, GuildConfig has method to do DB update. the cached object is updated directly, then the DB call is done.
Which mean we don't need to return the GuildConfig object after the call, we already have it when we did the DB call.

Adding this, would just add delay + create a nex GuildConfig object which wouldn't be useful.

However, the inconsistency you point out here is in InMemoryProvider.
updateGuild and updateAxon should both return a boolean.
I would also argue that there are useless method (methods that don't exist in other Providers) and useless async.
You can transform this PR in a PR that fix InMemoryProvider inconsistency that actually exist.

File: src/AxonClient.js [hunk 1]
```diff
@@ -323,6 +323,10 @@ class AxonClient extends EventEmitter {
      * @memberof AxonClient
      */
     async start() {
+        if (this._configs.webhooks.bot.token) {
```
Comment (Khaazz) [inline src/AxonClient.js]: Yeah no, this is not the correct way to do this.
there is nothing obligating the user to provide the token here.
+ this should be a separate PR.

**Comment by Khaazz** [comment]
So this is indeed better.
Now I didn't really realise when i first merged the PR adding this. But all `updateThing` methods beside `updateGuild` and `updateAxon` are never used internally and are not needed.
eg: https://github.com/Khaazz/AxonCore/blob/master/src/Database/InMemoryProvider.js#L39

I think it would be better to remove it as well.

File: src/AxonClient.js (hunk 1 above)
Comment (bsian03) [reply src/AxonClient.js]: Any other way to do this then?

**Comment by bsian03** [comment]
> I think it would be better to remove it as well.

Type boolean?

File: src/AxonClient.js (hunk 1 above)
Comment (Khaazz) [reply src/AxonClient.js]: For djs, we can easily make sure the token is provided. (since passed in djs libInterface).
For eris, it's directly added in eris client. We could manually check client.token, but they could have renamed the prop..
Maybe a timeout on `connect()`.
I'm thinking about it but I'm not sure yet.
Doing the djs check is easy though.

**Comment by Khaazz** [comment]
> > I think it would be better to remove it as well.
> 
> Type boolean?

No I really mean just deleting the methods.
Any methods that exist in InMemoryProvider and not in ADBProvider is not used internally and shouldn't exist in inMemoryProvider.

File: src/Database/InMemoryProvider.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ class InMemoryProvider extends ADBProvider {
         const guild = await this.fetchGuild(gID);
         guild[key] = value;
         this.saveGuildSchema(gID, guild);
```
Comment (Khaazz) [inline src/Database/InMemoryProvider.js]: This shouldn't call `this.saveGuildSchema` but should directly udpate the cache 
```suggestion
        this.axon.guildConfigs.set(gID, guild);
```
Ref: #53 that will change `saveGuildSchema` implementation to the correct one

File: src/Database/InMemoryProvider.js [hunk 3]
```diff
@@ -82,7 +82,7 @@ class InMemoryProvider extends ADBProvider {
             axonConf = await this.initAxon();
```
Comment (Khaazz) [inline src/Database/InMemoryProvider.js]: forgot here: this should call `fetchAxon` instead
